src/genius/family-form/family-form.component.ts

```diff
@@ -163,35 +228,52 @@ export class FamilyFormComponent {
     this.family.mother = person;
   }
 
-  addChild(person: Person): void {
-    this.family.children.push(person)
+  checkChild(index: number): Boolean {
+    return Object.keys(this.getChild(index)).length > 0
+  }
+
+  getCompleteChildrenAmount(): number {
+    let amount = 0;
+    if (this.getChildren()) {
+      this.getChildren().forEach(child => {
+        if (Object.keys(child).length > 0) amount++;
+      })
+    }
+
+    return amount;
   }
 
   getFamilyList() {
-    return this.dataProvider.getFamilies();
+    this.dataProvider.getFamilies().subscribe(res => this.families = res,
```

Comment:
name variable reasonable: res means 'result' or 'response' or something else ?
name it response